### ISSUE TITLE
Blocking of admissable diffracted paths?

### ISSUE THREAD

**Issue by johnrudge** [crotwell/TauP#18]
First off, thank you @crotwell for the taup toolkit! 

I've noted a change in behaviour from TauP 2.5.0 and 2.6.0 that seems to block diffracted paths that were once admissable. For example, on 2.5.0 one can do:
```
taup time -deg 220 -ph PedPdiffKP
```
```

Model: iasp91
Distance   Depth   Phase        Travel    Ray Param  Takeoff  Incident  Purist    Purist
  (deg)     (km)   Name         Time (s)  p (s/deg)   (deg)    (deg)   Distance   Name 
----------------------------------------------------------------------------------------
  140.00     0.0   PedPdiffKP   1506.17     4.439     13.39    13.39   220.00   = PedPdiffKP

```
Whereas on 2.6.0 the toolkit states
```
Warning: Error with phase=PedPdiffKP
TauP Exception: TauModel Exception: can't be downgoing as we are already below: 6 5
Skipping this phase
```
I think the problem lies in this commit:
https://github.com/crotwell/TauP/commit/b534c8ed4bd05c2d4608e81feee35825d56dfa35
which maybe too zealous in blocking allowable phases. I understand this was introduced to block phases like ScScS, and a similar bug report on this has been filed for obspy (https://github.com/obspy/obspy/issues/2774). I'm hoping to port your fix to obspy to correct the bad phases there, but I want to make sure I don't kill off phases that are ok.

**Comment by johnrudge** [comment]
I have the wrong commit here. It should be
https://github.com/crotwell/TauP/commit/99967206e2e69fecf227d9f9ade89a6bc3a98980

**Comment by crotwell** [comment]
Looking into this. I think the problem came from  062e05169c4e6acc5ece47b46f4f534d63d8b687

That was to try to deal with a model where the entire mantle had a negative velocity gradient, so no mantle turning rays. But tricky as you can get a reflection off of the core for the "last" ray parameter in this case and TauP was interpolating across a shadow zone. I think my fix worked for that particular test, but in a way that broke PedPdiffKP. This may be a case where the ray parameter needs to be restricted to >  instead of >= to avoid the pathological shadow zone, but I need to think about it some more.

Thanks for reporting this, hopefully I will be able to get a fix in soon.

**Comment by crotwell** [comment]
I think I have this fixed in ff98c737a2b1eeb3d28dff1b21fe474e56a92e4e

Can you verify?

If you have any other interesting phases that would be good to test for, let me know and I will add to the test case.

**Comment by johnrudge** [comment]
Yes, that does seem to fix it! Thanks for getting to this so fast! I'll let you know if I have any issues with any other funky phases.
